public/app/features/alerting/unified/components/silences/SilencesEditor.tsx

```diff
@@ -22,145 +21,140 @@ import {
   Input,
   LinkButton,
   LoadingPlaceholder,
+  Stack,
   TextArea,
   useStyles2,
 } from '@grafana/ui';
-import { alertSilencesApi } from 'app/features/alerting/unified/api/alertSilencesApi';
+import { alertRuleApi } from 'app/features/alerting/unified/api/alertRuleApi';
+import { alertSilencesApi, SilenceCreatedResponse } from 'app/features/alerting/unified/api/alertSilencesApi';
+import { MATCHER_ALERT_RULE_UID } from 'app/features/alerting/unified/utils/constants';
 import { getDatasourceAPIUid } from 'app/features/alerting/unified/utils/datasource';
-import { Matcher, MatcherOperator, Silence, SilenceCreatePayload } from 'app/plugins/datasource/alertmanager/types';
+import { MatcherOperator, SilenceCreatePayload } from 'app/plugins/datasource/alertmanager/types';
 
-import { useURLSearchParams } from '../../hooks/useURLSearchParams';
 import { SilenceFormFields } from '../../types/silence-form';
-import { matcherFieldToMatcher, matcherToMatcherField } from '../../utils/alertmanager';
-import { parseQueryParamMatchers } from '../../utils/matchers';
+import { matcherFieldToMatcher } from '../../utils/alertmanager';
 import { makeAMLink } from '../../utils/misc';
 
 import MatchersField from './MatchersField';
 import { SilencePeriod } from './SilencePeriod';
 import { SilencedInstancesPreview } from './SilencedInstancesPreview';
+import { getDefaultSilenceFormValues, getFormFieldsForSilence } from './utils';
 
-interface Props {
-  silenceId?: string;
+interface ConnectedSilenceEditorProps {
+  silenceId: string;
   alertManagerSourceName: string;
 }
 
-const defaultsFromQuery = (searchParams: URLSearchParams): Partial<SilenceFormFields> => {
-  const defaults: Partial<SilenceFormFields> = {};
+/**
+ * Silences editor for editing an existing silence.
+ *
+ * Fetches silence details from API, based on `silenceId`
+ */
+export const ConnectedSilencesEditor = ({ silenceId, alertManagerSourceName }: ConnectedSilenceEditorProps) => {
```

Comment:
How about renaming to `ExistingSilenceEditor`?
`Connected` remind me redux connected components which is not true in this case 😄